library/core/src/iter/traits/iterator.rs

```diff
@@ -4060,6 +4060,40 @@ pub trait Iterator {
     {
         unreachable!("Always specialized");
     }
+
+    /// Returns `true` if the iterator contains a value.
+    ///
+    /// 'contains' is short-circuiting; in other words, it will stop processing
+    /// as soon as the function finds the item in the `Iterator`.
+    ///
+    /// Performance:
+    /// This method checks the whole iterator, which takes O(n) time.
+    /// If the iterator is sorted, or a hash map please use the appropriate method instead.
+    ///
+    /// Example:
+    /// ```
+    /// #![feature(iter_contains)]
+    /// assert!([1, 2, 3].iter().contains(&1));
+    /// assert!(![1, 2, 3].iter().contains(&4));
+    /// // You can check if a String is in a string slice iterator.
+    /// assert!(["a", "b", "c"].iter().contains(&"b".to_owned()));
+    /// // You can also check if a String iterator contains a string slice.
+    /// assert!(["a".to_owned(), "b".to_owned(), "c".to_owned()].iter().contains(&"b"));
```

Comment:
Maybe make it an example that you couldn't just use `slice::contains(...)` on, like `[1, 2, 3].iter().map(|x| x * 3).contains(6)`